command/login_test.go

```diff
@@ -454,64 +501,83 @@ func TestLoginCommand_Run(t *testing.T) {
 			t.Errorf("expected %d to be %d", code, exp)
 		}
 
+		expected := methodID
+		output = ui.OutputWriter.String() + ui.ErrorWriter.String()
+		t.Logf("\n%+v", output)
+		if !strings.Contains(output, expected) {
+			t.Fatalf("expected stored token: %q, got: %q", expected, output)
+		}
+
 		tokenHelper, err := cmd.TokenHelper()
 		if err != nil {
 			t.Fatal(err)
 		}
 		storedToken, err := tokenHelper.Get()
+		if storedToken != "" {
+			t.Fatal("expected empty stored token")
+		}
 		if err != nil {
 			t.Fatal(err)
 		}
-		output = ui.OutputWriter.String() + ui.ErrorWriter.String()
-		t.Logf("\n%+v", output)
-		if !strings.Contains(output, storedToken) {
-			t.Fatalf("expected stored token: %q, got: %q", storedToken, output)
-		}
 	})
 
-	t.Run("login_mfa_two_phase", func(t *testing.T) {
+	t.Run("login_mfa_two_phase_non_interactive_method_name", func(t *testing.T) {
 		t.Parallel()
 
 		client, closer := testVaultServer(t)
 		defer closer()
 
 		ui, cmd := testLoginCommand(t)
 
-		userclient, entityID, methodID := testhelpers.SetupLoginMFATOTP(t, client)
+		userclient, entityID, methodID, methodName := testhelpers.SetupLoginMFATOTP(t, client)
 		cmd.client = userclient
 
-		_ = testhelpers.RegisterEntityInTOTPEngine(t, client, entityID, methodID)
+		engineName := testhelpers.RegisterEntityInTOTPEngine(t, client, entityID, methodID)
 
 		// clear the MFA creds just to be sure
 		cmd.client.SetMFACreds([]string{})
 
 		code := cmd.Run([]string{
 			"-method", "userpass",
+			"-non-interactive",
 			"username=testuser1",
 			"password=testpassword",
 		})
 		if exp := 0; code != exp {
 			t.Errorf("expected %d to be %d", code, exp)
 		}
 
-		expected := methodID
 		output = ui.OutputWriter.String() + ui.ErrorWriter.String()
-		t.Logf("\n%+v", output)
-		if !strings.Contains(output, expected) {
-			t.Fatalf("expected stored token: %q, got: %q", expected, output)
-		}
 
-		tokenHelper, err := cmd.TokenHelper()
+		reqIdReg, err := regexp.Compile(`mfa_request_id\s+(?P<name>[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12})?\s+mfa_constraint.*`)
```

Comment:
You can use MustCompile to avoid the if.

Reply:
```suggestion
		reqIdReg := regexp.MustCompile(`mfa_request_id\s+([0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12})\s+mfa_constraint`)
```

I think this simpler regexp should work?  No need for <name> since you're not using that; no reason to make the uuid optional; no need for trailing `.*`.